base/features/aws-api-gateway-graal/skeleton/Dockerfile

```diff
@@ -27,9 +27,9 @@ CMD ["/usr/lib/graalvm/bin/native-image"]
 FROM graalvm
 COPY --from=builder /home/application/ /home/application/
 WORKDIR /home/application
-RUN /usr/lib/graalvm/bin/native-image --no-server -cp @jarPath@ 
+RUN /usr/lib/graalvm/bin/native-image --initialize-at-build-time=com.amazonaws.serverless.proxy.model --no-server -cp @jarPath@
```

Comment:
Instead of including that in the profile it is better if you include it directly in Micronaut code, so that configuration is included in the jars we release.
Take a look at how I did it for micronaut-micrometer: https://github.com/micronaut-projects/micronaut-micrometer/commit/6ad130b3a2fd429b5ab92f44101bc7f9a201a179

Reply:
I'm not 100% sure where to make the change.

When I use the cli to create the app it creates the native-image.properties file from this profile.

This is the contents of that file with the fix included (the last line is the fix):

```
Args = -H:IncludeResources=application.yml|log4j2.xml \
       -H:Name=server \
       -H:-AllowVMInspection \
       -H:Class=io.micronaut.function.aws.runtime.MicronautLambdaRuntime \
       --initialize-at-build-time=com.amazonaws.serverless.proxy.model
```

I looked in the micronaut-aws repo: https://github.com/micronaut-projects/micronaut-aws
and couldn't figure out where to include that line. Please advise.

Reply:
You need to create the file `resources/META-INF/native-image/io.micronaut.function.aws.proxy/native-image.properties` in https://github.com/micronaut-projects/micronaut-aws/tree/master/function-aws-api-proxy/src/main directory and then add the configuration you added to the Dockerfile:

```
Args = --initialize-at-build-time=com.amazonaws.serverless.proxy.model
```

That file will be included in Micronaut's jar and then when generating the native-image Graal will process all the `native-image.properties` files in the jars, so we don't need to modify the `Dockerfile` because the support is included in Micronaut itself.